### DISCUSSION TITLE
How to undestand query language

### DISCUSSION THREAD

**Discussion by eankit** [tree-sitter/tree-sitter-scala#183] [General]
I want to write queries using query language for scala code but I am finding it difficult to read from grammer files,
do we have any sample query code or a few documentations which help me understand query language so that I can write queries for scala and leverage tree sitter scala to the max. 
Do we have something similar to this for Scala
https://tree-sitter.github.io/tree-sitter/playground

**Comment by ckipp01** [comment]
> do we have any sample query code or a few documentations which help me understand query language

Hey @eankit, to get a better understanding I recommend digging a bit into the docs. There is a pretty large section on [using parsers](https://tree-sitter.github.io/tree-sitter/using-parsers) that may give you some insight. You can also take a look at our [corpus](https://github.com/tree-sitter/tree-sitter-scala/tree/master/corpus) files to get an idea of how things get parsed.

> Do we have something similar to this for Scala
https://tree-sitter.github.io/tree-sitter/playground

Depending on your editor choice, many of us use https://github.com/nvim-treesitter/playground for this.
